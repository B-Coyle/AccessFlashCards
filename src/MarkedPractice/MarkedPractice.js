import React, {Component} from 'react';
import './MarkedPractice.scss';

export default class MarkedPractice extends Component {
    constructor(props) {
        super(props);
        this.state = {
            practice: false,
            good: false,
            checked: false,
            studyMaterials: JSON.parse(localStorage.getItem('needPractice')),
            newQuestion: []
        }
    }
    displayGood=()=>{
        this.setState({
            practice: true
        })
    }

    displayNextPractice=()=>{
        let practiceQuestion = this.state.studyMaterials.practice.sort(() => .5 - Math.random()).pop();
        this.setState ({
        newQuestion: practiceQuestion,
        checked: true,
        good: true
        })
    }

    restartPracticeGame=()=>{
        this.props.restartGame()
    }

    flipPracticeCard=()=>{
        this.props.flipCard()
    }

render() {
console.log('test', this.state.newQuestion)
    if(this.state.practice === false && this.state.good === false) {
        return (
        <article className="endSelectBtn">
        <h2>What would you like to review?</h2>
        <input type="button" value="Need more practice" onClick={this.displayNextPractice} />
        <input type="button" value="Good to Go" onClick={this.displayGood}/>
        <input type="button" value="Take me to the Start page" onClick={this.restartPracticeGame}/>
        </article>
    )
    } else {
    if(this.props.isFlipped === true) {
        return (
            <section className="quizCard">
            <article className="answerCard">
                <h2>Practice Deck</h2>
                <h3 className="category">Category: </h3>
                <h4>{this.state.newQuestion.category}</h4>
                <h3 className="answerLabel">Answer:</h3>
                <h4>{ this.state.newQuestion.answer}</h4>
            </article>
            <article className="cardBtnContainer">
                <input type='button' className="flipBtn" value="Flip Card" onClick={this.flipCard}/>
                <input type='button' className="nextBtn" value='Next Card' onClick={this.displayNextPractice} />
            </article> 
            </section>
        )
    } else {
        return (
    <section className="quizCard" >
        <article className="questionCard" >
            <h2>Practice Deck</h2>
            <h3 className="category"><span className="categorySpan">Category: </span></h3>
                <h4>{this.state.newQuestion.category}</h4>
            <h3 className="questionLabel"><span className="questionSpan">Question: </span></h3>
                <h4>{ this.state.newQuestion.question}</h4>
        </article>
        <article className="buttonContainer">
            <input type='button' className="flipBtn" value="Flip Card" onClick={this.props.flipCard}/>
            <input type='button' className="nextBtn" value='Next Card'  onClick={this.displayNextPractice}/>
        </article>
    </section>)
            }
        }
    }
}