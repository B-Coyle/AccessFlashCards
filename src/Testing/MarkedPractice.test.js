import React from 'react';
import MarkedPractice from '../MarkedPractice/MarkedPractice';
import { shallow } from 'enzyme'; 
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const mockDisplayPractice = jest.fn()
const mockCheckIfFlipped = jest.fn()
const mockStateFlipped = false
const mockData = []

describe('MarkedPractice', () => {
let wrapper;

beforeEach(() => {
    wrapper= shallow(
        <MarkedPractice 
            isFlipped = {mockStateFlipped}
            displayNextCard = {mockDisplayPractice}
            flipCard= {mockCheckIfFlipped}
        /> 
    )
});
it('should have a default state', () => {
    
})

it('should match rendered snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an event listener on the click start button', () => {
    wrapper.find('article').childAt(1).simulate('click')
    expect(mockDisplayPractice).toHaveBeenCalled();
});


})