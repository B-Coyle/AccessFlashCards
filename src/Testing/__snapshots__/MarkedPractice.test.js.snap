// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkedPractice should match rendered snapshot 1`] = `
<article
  className="endSelectBtn"
>
  <h2>
    What would you like to review?
  </h2>
  <input
    className="practiceBtn"
    onClick={[Function]}
    type="button"
    value="Need more practice"
  />
  <input
    className="goodBtn"
    onClick={[Function]}
    type="button"
    value="Good to Go"
  />
  <input
    className="resetPracticeBtn"
    onClick={[Function]}
    type="button"
    value="Take me to the Start page"
  />
</article>
`;
